packages/flutter/lib/src/cupertino/scaffold.dart

```diff
@@ -0,0 +1,212 @@
+  /// Construct a [CupertinoScaffold] without tabs.
+  // TODO(xster): document that page transitions will happen behind the navigation
+  // bar.
+  const CupertinoScaffold({
+    Key key,
+    this.navigationBar,
+    @required this.child,
+  })
+      : tabBar = null,
+        rootTabPageBuilder = null,
+        super(key: key);
+
+  /// Construct a [CupertinoScaffold] with tabs.
+  ///
+  /// A [tabBar] and a [rootTabPageBuilder] are required. The [CupertinoScaffold]
+  /// will automatically listen to the provided [CupertinoTabBar]'s tap callbacks
+  /// to change the active tab.
+  ///
+  /// Tabs' contents are built with the provided [rootTabPageBuilder] at the active
+  /// tab index. [rootTabPageBuilder] must be able to build the same number of
+  /// pages as the [tabBar.items.length]. Inactive tabs will be moved [Offstage]
+  /// and its animations disabled.
+  const CupertinoScaffold.tabbed({
+    Key key,
+    this.navigationBar,
+    @required this.tabBar,
+    @required this.rootTabPageBuilder,
+  })
+      : assert(tabBar != null),
+        assert(rootTabPageBuilder != null),
+        child = null,
+        super(key: key);
+
+  /// The [navigationBar], typically a [CupertinoNavigationBar], is drawn at the
+  /// top of the screen.
+  ///
+  /// If translucent, the main content may slide behind it.
+  /// Otherwise, the main content's top margin will be offset by its height.
+  // TODO(xster): document its page transition animation when ready
+  final PreferredSizeWidget navigationBar;
+
+  /// The [tabBar] is a [CupertinoTabBar] drawn at the bottom of the screen
+  /// that lets the user switch between different tabs in the main content area
+  /// when present.
+  ///
+  /// When provided, [CupertinoTabBar.currentIndex] will be ignored and will
+  /// be managed by the [CupertinoScaffold] to show the currently selected page
+  /// as the active item index. If [CupertinoTabBar.onTap] is provided, it will
+  /// still be called. [CupertinoScaffold] automatically also listen to the
+  /// [CupertinoTabBar]'s `onTap` to change the [CupertinoTabBar]'s `currentIndex`
+  /// and change the actively displayed tab in [CupertinoScaffold]'s own
+  /// main content area.
+  ///
+  /// If translucent, the main content may slide behind it.
+  /// Otherwise, the main content's bottom margin will be offset by its height.
+  final CupertinoTabBar tabBar;
+
+  /// An [IndexedWidgetBuilder] that's called when tabs become active. Used
+  /// when a tabbed scaffold is constructed via the [new CupertinoScaffold.tabbed]
+  /// constructor.
+  ///
+  /// Content can slide under the [navigationBar] or the [tabBar] when they're
+  /// translucent.
+  final IndexedWidgetBuilder rootTabPageBuilder;
+
+  /// Widget to show in the main content area when the scaffold is used without
+  /// tabs.
+  ///
+  /// Content can slide under the [navigationBar] or the [tabBar] when they're
+  /// translucent.
+  final Widget child;
```

Comment:
the docs for child and rootTabPageBuilder should clarify that they can't both be non-null, that one must be non-null, and they should each point to the relevant constructor.